src/main/java/org/jabref/gui/preview/PreviewViewer.java

```diff
@@ -137,23 +145,26 @@ public void setTheme(Theme theme) {
     }
 
     private void highlightSearchPattern() {
+        String callbackForUnmark = "";
         if (searchHighlightPattern.isPresent()) {
-            String pattern = searchHighlightPattern.get().pattern();
-
-            previewView.getEngine().executeScript(
-                    "var markInstance = new Mark(document.getElementById(\"content\"));" +
-                            "markInstance.unmark({" +
-                            "  done: function(){" +
-                            "    markInstance.markRegExp(/" + pattern.replaceAll("(?<!\\\\)/", "\\\\/") + "/gmi);" +
-                            "    }" +
-                            "  });"
-            );
-        } else {
-            previewView.getEngine().executeScript(
-                    "var markInstance = new Mark(document.getElementById(\"content\"));" +
-                            "markInstance.unmark()"
-            );
+            String javaScriptRegex = createJavaScriptRegex(searchHighlightPattern.get());
+            callbackForUnmark = String.format(JS_MARK_REG_EXP_CALLBACK, javaScriptRegex);
         }
+        String unmarkInstance = String.format(JS_UNMARK_WITH_CALLBACK, callbackForUnmark);
+        previewView.getEngine().executeScript(unmarkInstance);
+    }
+
+    /**
+     * Returns the String representation of a JavaScript regex object. The method does not take into account differences between the regex implementations in Java and JavaScript.
+     *
+     * @param regex Java regex to print as a JavaScript regex
+     * @return JavaScript regex object
+     */
+    private static String createJavaScriptRegex(Pattern regex) {
+        String pattern = regex.pattern();
+        // Create a JavaScript regex using the forward slash pattern
+        pattern = UNESCAPED_FORWARD_SLASH.matcher(pattern).replaceAll("\\\\/");
+        return "/" + pattern + "/gmi";
```

Comment:
@koppor 
> Would it be possible to extract this as constant (and thus documenting why this replacement is done)?

I noticed that I misunderstood you. I have extracted the String constants (as well) and hopefully improved readability. I don't think I should be more explicit regarding the forward replacements, unless I also link to how to use a JavaScript regex literal (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Regular_Expressions). Perhaps I should add this link?

Reply:
Yes, I like links in the code. Even if there are opinions that they might oudate. Neverthless, 1 good link is better than 0 links

Reply:
Well, I compromised and added the link I believe is the least likely to get outdated 😛